src/fal/cli/fal_runner.py

```diff
@@ -55,24 +57,47 @@ def fal_run(args: argparse.Namespace):
 
     scripts = _select_scripts(args, models, faldbt)
 
+    global_scripts = _get_global_scripts(faldbt, args.before)
+
     if args.before:
         if not _scripts_flag(args):
             # run globals when no --script is passed
-            _run_global_scripts(faldbt, args.before)
+            _run_scripts(args, global_scripts, faldbt)
 
-        results = run_scripts(scripts, faldbt)
-        raise_for_run_results_failures(scripts, results)
+        _run_scripts(args, scripts, faldbt)
 
     else:
-        results_after = run_scripts(scripts, faldbt)
-        raise_for_run_results_failures(scripts, results_after)
+        _run_scripts(args, scripts, faldbt)
 
-        results_post_hook = run_scripts(_get_all_post_hooks(models, faldbt), faldbt)
-        raise_for_run_results_failures(scripts, results_post_hook)
+        post_hook_scripts = _get_all_post_hooks(models, faldbt)
+        _run_scripts(args, post_hook_scripts, faldbt)
 
         if not _scripts_flag(args):
             # run globals when no --script is passed
-            _run_global_scripts(faldbt, args.before)
+            _run_scripts(args, global_scripts, faldbt)
+
+
+def _run_scripts(args: argparse.Namespace, scripts: List[FalScript], faldbt: FalDbt):
+    scheduler = Scheduler(
+        [
+            TaskGroup(
+                FalHookTask(
+                    hook_path=script.path,
+                    bound_model=script.model,
+                    is_post_hook=script.is_post_hook,
+                )
+            )
+            for script in scripts
+        ]
+    )
```

Comment:
should we create an alternative constructor for the scheduler that takes script as an input?

Reply:
I did something similar, but not quite